public/docs/_examples/cb-ts-to-js/js-es6/app/hero-di-inject-additional.component.es6

```diff
@@ -0,0 +1,61 @@
+import {
+  Attribute,
+  Component,
+  Inject,
+  Optional,
+  NgModule
+} from '@angular/core';
+import { BrowserModule } from '@angular/platform-browser';
+
+// #docregion
+class TitleComponent {
+  constructor(titlePrefix, title) {
+    this.titlePrefix = titlePrefix;
+    this.title  = title;
+    this.msg = '';
+  }
+
+  ok() {
+    this.msg = 'OK!';
+  }
+}
+
```

Comment:
Since Functions are hoisted, it might be nice to move the annotations assignment above the actual constructor that it is referencing. This would keep the "shape" of the code more consistent between the various versions.